sdks/cpp/connections/gRPC/examples/status_update/status_update.cpp

```diff
@@ -117,13 +117,16 @@ void statusUpdateExample(){
              * Protobuf lite does not support converting messages to JSON strings.
              * @todo: Implement a toString method for catena values.
              */
-            std::cout << p->getOid() << " has been updated" << '\n';
+            std::cout << p->getOid() << " has been changed by client" << '\n';
         });
         Param<int32_t>& aNumber = *dynamic_cast<Param<int32_t>*>(dm.getItem("/counter", Device::ParamTag{}));
         while (globalLoop) {
             std::this_thread::sleep_for(std::chrono::seconds(1));
-            Device::LockGuard lg(dm); 
-            aNumber.get()++;
+            {
+                Device::LockGuard lg(dm); 
+                aNumber.get()++;
+            }
+            dm.valueSetByServer.emit("/counter", &aNumber, 0);
```

Comment:
this is a problem - aNumber could be updated by another thread so it should be called within the scope of the lock guard. Of course this means that any recipients of the signal must not try to apply the lock, and they should be minimize the time spent in the function call.